sql/rowexec/dml_iters.go

```diff
@@ -33,6 +33,47 @@ type triggerRollbackIter struct {
 	savePointName string
 }
 
+func containsTrigger(node sql.Node) bool {
```

Comment:
This is a good use case for query flags. When we build a trigger in planbuilder mark a trigger use, and then read that back instead of traversing the node.